From 0c060512c1bf6719391e2d3351c8cb757bec29cc Mon Sep 17 00:00:00 2001
From: Gabriel Somlo <gsomlo@gmail.com>
Date: Thu, 23 Jan 2025 07:53:52 -0500
Subject: [PATCH] Fix undefined type error in 3rdparty/json11/json11.cpp
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Under certain conditions (e.g., building on Fedora 42
using gcc-15.0.1), compilation fails with the following
error:

    "error: ‘uint8_t’ does not name a type"

Explicitly include <cstdint> to prevent that situation.

Signed-off-by: Gabriel Somlo <gsomlo@gmail.com>
---
 3rdparty/json11/json11.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/3rdparty/json11/json11.cpp b/3rdparty/json11/json11.cpp
index a0fcf866cc..b6c95e0447 100644
--- a/3rdparty/json11/json11.cpp
+++ b/3rdparty/json11/json11.cpp
@@ -23,6 +23,7 @@
 #include <cassert>
 #include <cmath>
 #include <cstdlib>
+#include <cstdint>
 #include <cstdio>
 #include <climits>
 #include <cerrno>
